volumes/miovision/sql/create-table-volumes_15min.sql

```diff
@@ -3,66 +3,70 @@
 -- DROP TABLE miovision_api.volumes_15min;
 CREATE TABLE miovision_api.volumes_15min
 (
-  volume_15min_uid serial NOT NULL,
-  intersection_uid integer,
-  datetime_bin timestamp without time zone,
-  classification_uid integer,
-  leg text,
-  dir text,
-  volume numeric,
-  CONSTRAINT volumes_15min_pkey PRIMARY KEY (volume_15min_uid),
-  CONSTRAINT volumes_15min_intersection_uid_datetime_bin_classification__key UNIQUE (intersection_uid, datetime_bin, classification_uid, leg, dir)
+    volume_15min_uid serial NOT NULL,
+    intersection_uid integer,
+    datetime_bin timestamp without time zone,
+    classification_uid integer,
+    leg text,
+    dir text,
+    volume numeric,
+    CONSTRAINT volumes_15min_intersection_uid_datetime_bin_classification_pkey
```

Comment:
Do you think we should make volume_15min_uid the primary key, and make this chunk a unique constraint instead?

Reply:
to make foreign references we need a the column (volumes_15min_uid or volumes_15min_mvt) to have a unique constraint/ as the primary key. I think our original table structure has volumes_15min_uid as the primary key and a unique constraint on the list of columns.

Reply:
"To create a unique or primary key constraint on a partitioned table, the partition keys must not include any expressions or function calls and the **constraint's columns must include all of the partition key columns**." [Source](https://www.postgresql.org/docs/current/ddl-partitioning.html#DDL-PARTITIONING-DECLARATIVE-LIMITATIONS)
The other option would be to add the uid to the current Pkey.

Reply:
Ok lets add it to the primary key then. I was initially thinking we can make it a unique constraint since we would not want duplicate uids, as serial column could produce duplicate uid if we manually specify a value in the serial column (thought now that i think about it, we probably will never do this)

Reply:
I just added these lines: 
```sql
--add volume_15min_mvt_uid to pkey so it can be used in fkey
ALTER TABLE miovision_api.volumes_15min_mvt DROP CONSTRAINT volumes_15min_mvt_int_uid_dt_bin_class_leg_mvmt_uid_pkey;
ALTER TABLE miovision_api.volumes_15min_mvt ADD CONSTRAINT volumes_15min_mvt_int_uid_dt_bin_class_leg_mvmt_uid_pkey
PRIMARY KEY (volume_15min_mvt_uid, intersection_uid, datetime_bin, classification_uid, leg, movement_uid);
```

Reply:
thanks! running

Reply:
ok so im still running into the same error 
`ERROR:  there is no unique constraint matching given keys for referenced table "volumes_15min_mvt"`
I believe its because we are referencing `volume_15min_mvt_uid` and not `volumes_15min_mvt_int_uid_dt_bin_class_leg_mvmt_uid_pkey`, where the first one technically on itself can be not unique, it just needs to be unique with combination of another columns. Can we make that one column a unique constraint + cascade to the rest of the partition?

Reply:
Let's try again! I think the declarative partitioning takes care of cascading to the partitions. Looks to be the case for the change we made above in this thread.
```sql
ALTER TABLE miovision_api.volumes_15min_mvt ADD CONSTRAINT volume_15min_mvt_uid_unique UNIQUE (volume_15min_mvt_uid);
```

Reply:
sigh
```
ERROR:  unique constraint on partitioned table must include all partitioning columns
DETAIL:  UNIQUE constraint on table "volumes_15min_mvt" lacks column "datetime_bin" which is part of the partition key.
```

Reply:
seems like we cannot have a foreign key referencing the column `volumes_15min_mvt` alone in this case, as it has to include partition key. Unless we reference datetime_bin + volumes_15min_mvt, however this wouldn't work cause  datetime_bin in volumes_15min_mvt wont be the same as the datetime_bin in `volumes`.  
I think not having foreign key is not too big of a deal, and saving a little space from not creating unique constraint+ index on this table is not a bad thing. 
Maybe in the future we can revisit the use of volumes_15min_mvt_uid(?) 
I'm leaning towards dropping the foreign key issue and merge this PR. Thoughts?

Reply:
I agree removing the foreign key relationship on `volumes` referencing `volumes_15min_mvt` is fine since it's incompatible with the new partition design. The new replacement index will help with this condition: 
https://github.com/CityofToronto/bdit_data-sources/blob/99a0e2c1861a76b0650111b2d3e925be5950e5d7/volumes/miovision/sql/function-aggregate-volumes_15min_mvt.sql#L62C1-L63C43